bindings/nodejs/lib/types/output.ts

```diff
@@ -0,0 +1,241 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { UnlockCondition } from './unlock_condition';
+import { Feature } from './feature';
+import { HexEncodedString, INativeToken, TokenSchemeTypes } from '@iota/types';
+
+/**
+ * All of the output types.
+ */
+enum OutputType {
+    Treasury = 2,
+    Basic = 3,
+    Alias = 4,
+    Foundry = 5,
+    Nft = 6,
+}
+
+abstract class Output /*implements ICommonOutput*/ {
+    private amount: string;
+    private type: OutputType;
+
+    constructor(type: OutputType, amount: string) {
+        this.type = type;
+        this.amount = amount;
+    }
+    /**
+     * The type of output.
+     */
+    getType(): OutputType {
+        return this.type;
+    }
+
+    /**
+     * The amount of IOTA coins to held by the output.
```

Comment:
```suggestion
     * The amount of the output.
```
not necessarily IOTA, in Rust we also just use amount